packages/mask/popups/components/TokenPicker/index.tsx

```diff
@@ -20,34 +22,50 @@ const useStyles = makeStyles()((theme) => {
         sidebar: {
             paddingRight: theme.spacing(1),
         },
-        list: {
-            overflow: 'auto',
-            width: '100%',
-        },
     }
 })
 
 export interface TokenPickerProps extends Omit<BoxProps, 'onSelect'>, Pick<TokenItemProps, 'onSelect'> {
     defaultChainId?: ChainId
     chainId?: ChainId
     address?: string
+    chains?: ChainId[]
+    /** okx provides their own token list */
+    assetSource?: 'standard' | 'okx'
+    /** Do not allow to select other chains */
+    lockChainId?: boolean
     onChainChange?: (chainId: Web3Helper.ChainIdAll | undefined) => void
 }
 
 export const TokenPicker = memo(function TokenPicker({
     defaultChainId,
-    chainId,
+    chainId: propChainId,
     address,
+    assetSource = 'standard',
+    lockChainId = false,
+    chains,
     className,
     onSelect,
     onChainChange,
     ...rest
 }: TokenPickerProps) {
     const { classes, cx } = useStyles()
-    const [assets] = useFungibleAssets(NetworkPluginID.PLUGIN_EVM, undefined, {
+    const [sidebarChainId, setSidebarChainId] = useState<Web3Helper.ChainIdAll | undefined>(defaultChainId)
+    const chainId = (sidebarChainId || propChainId || defaultChainId) as ChainId
+    const [standardAssets] = useFungibleAssets(NetworkPluginID.PLUGIN_EVM, undefined, {
         chainId,
     })
-    const [sidebarChainId, setSidebarChainId] = useState<Web3Helper.ChainIdAll | undefined>(defaultChainId)
+    const { data: okxTokens } = useOKXTokenList(chainId, assetSource === 'okx')
+    const okxAssets = useMemo(() => {
+        if (!okxTokens?.length) return EMPTY_LIST
+        const balanceMap = new Map(standardAssets.map((x) => [x.address.toLowerCase(), x.balance]))
+        // To reduce queries, get balance from standardAssets and patch okxTokens with it
+        return okxTokens.map((x) => {
+            const balance = balanceMap.get(x.address.toLowerCase())
+            return !balance || balance === '0' ? x : { ...x, balance }
+        }) as typeof okxTokens
+    }, [okxTokens, standardAssets])
```

Comment:
```suggestion
       return okxTokens.map(x => {
           standardAssets.find(y => isSameAddress(x.address, y.address) && y.balance ? { ...x, balance } : x)
       })
```

Reply:
Nesting loops, that would be `O(n^2)`, `Map.get` is more efficient.

Reply:
There could be thousands of tokens in OKX's token list, so performance woud be a thing here.